src/main/scala/mesosphere/marathon/core/instance/update/InstanceUpdateOpResolver.scala

```diff
@@ -32,23 +25,23 @@ private[marathon] class InstanceUpdateOpResolver(
     * violated the future will fail with an [[IllegalStateException]], otherwise the operation
     * will be applied and result in an [[InstanceUpdateEffect]].
     */
-  def resolve(op: InstanceUpdateOperation)(implicit ec: ExecutionContext): Future[InstanceUpdateEffect] = {
+  def resolve(instancesBySpec: InstancesBySpec, op: InstanceUpdateOperation): InstanceUpdateEffect = {
```

Comment:
do we need instancesBySpec here? I think Option[Instance] would be enough - that would be the instance of that instance id that already exists (if exists)

Reply:
The error is thrown in the resolver for a non-existing instance. I did not want to move too much logic.

Reply:
I don't see how that is connected. If you have Option[Instance] you can still throw that error. I think doing that is ok and it's not moving too much logic.

Reply:
Oh, you mean we would pass a `maybeApp`? I'll follow up this PR.